core/src/main/scala/org/apache/spark/scheduler/HealthTracker.scala

```diff
@@ -157,8 +159,13 @@ private[scheduler] class HealthTracker (
     allocationClient match {
       case Some(a) =>
         logInfo(msg)
-        a.killExecutors(Seq(exec), adjustTargetNumExecutors = false, countFailures = false,
-          force = true)
+        if (!decommission) {
```

Comment:
Instead of `negative if` statement, shall we simplify by switch the block (line 166~167) and the block (line 163 ~ 164)?